### DISCUSSION TITLE
Explanation of the ANN to SNN Conversion Mechanism

### DISCUSSION THREAD

**Discussion by arkapravag22** [synsense/sinabs#203] [Q&A] [answered]
Hi, I am converting an ANN to SNN. I wished to know the conversion mechanism. Is it just replacing the ReLU with a spiking layer? In that case it would be necessary to manually tune the conversion by experimenting with different parameters like minimum threshold, reset and spikes at each timestep.

In other words is Sinabs doing some mechanism of its own by adjusting the parameters or it just adds a spiking layer with the parameters we specify?

**Comment by bauerfe** [comment, marked as answer]
You are correct. ReLU layers are replaced by spiking layers. Other than, parameters and the architecture of the network remain untouched. 

There is no automatic normalization of the weights, so you generally have to scale either the weights or the firing thresholds manually. See [here](https://sinabs.readthedocs.io/en/main/tutorials/weight_scaling.html) for more details.
